cumulus/parachains/runtimes/coretime/coretime-rococo/src/coretime.rs

```diff
@@ -0,0 +1,240 @@
+// Copyright 2022 Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::*;
+use codec::{Decode, Encode};
+use frame_support::{
+	parameter_types,
+	traits::{
+		fungible::{Balanced, Credit},
+		OnUnbalanced,
+	},
+};
+use pallet_broker::{CoreAssignment, CoreIndex, CoretimeInterface, PartsOf57600};
+use parachains_common::{AccountId, Balance, BlockNumber};
+use xcm::latest::prelude::*;
+
+pub struct CreditToCollatorPot;
+impl OnUnbalanced<Credit<polkadot_core_primitives::AccountId, Balances>> for CreditToCollatorPot {
+	fn on_nonzero_unbalanced(credit: Credit<polkadot_core_primitives::AccountId, Balances>) {
+		let staking_pot = CollatorSelection::account_id();
+		let _ = <Balances as Balanced<_>>::resolve(&staking_pot, credit);
+	}
+}
+
+/// A type containing the encoding of the Broker pallet in the Relay chain runtime. Used to
+/// construct any remote calls. The codec index must correspond to the index of `Broker` in the
+/// `construct_runtime` of the Relay chain.
+#[derive(Encode, Decode)]
+enum BrokerRuntimePallets {
+	#[codec(index = 74)]
+	Broker(CoretimeProviderCalls),
+}
+
+/// Call encoding for the calls needed from the Broker pallet.
+#[derive(Encode, Decode)]
+enum CoretimeProviderCalls {
+	#[codec(index = 1)]
```

Comment:
I assume this pallet does not exist yet? should not we start from 0?

Reply:
It was a coincidence from the order of the methods in the [Coretime Interface](https://github.com/paritytech/polkadot-sdk/blob/joe-coretime-chain/substrate/frame/broker/src/coretime_interface.rs#L53). Starting indexing from 0 leads to this one being index 1.

Reply:
usually it starts from 0, but it does not concern this pr

Reply:
That's what I mean though, it started at 0 with `latest`, but I just don't need to use it so I haven't included it here, the next is `request_core_count` which is then index 1, and so on